cpp/support/csr_array.h

```diff
@@ -45,6 +47,10 @@ class CSRArray {
   /*! \brief Get the number of rows in the CSRArray. */
   int32_t Size() const { return static_cast<int32_t>(indptr_.size()) - 1; }
 
+  friend std::size_t SizeOfHeap(const CSRArray<DataType>& arr) {
```

Comment:
```suggestion
  friend std::size_t MemorySize(const CSRArray<DataType>& arr) {
```

SizeOfHeap -> MemorySize for clarity. ditto for other places